### ISSUE TITLE
Running Cells automatically scrolls the notebook to the bottom

### ISSUE THREAD

**Issue by matthewgasbarro** [jupyterlab/jupyterlab#16402]
## Description

Notebooks that have lots of cells are auto-scrolled to the last cell output when running all cells, despite where the cursor is on the notebook, or what is selected.

## Reproduce

1. Populate a notebook with enough cells such that all the cells plus their outputs would necessitate scrolling the notebook downwards to view the last output
2. Select the top-most cell
3. Click Run -> Run All Cells (or Run Selected Cell and Below)
4. Notice the notebook auto-scrolls to the bottom of the very last cell output

## Expected behavior

I would expect there would be no scrolling and the user scrolls to the bottom if they so choose, or there is a setting to enable/disable auto-scrolling.

## Context

JupyterLab 4.2.1
Can be reproduced on https://jupyter.org/try-jupyter/lab/ after adding or duplicating the cells to make the notebook more lengthy.

**Comment by krassowski** [comment]
I agree. Cross-referencing https://github.com/jupyterlab/jupyterlab/issues/9283

**Comment by Meriem-BenIsmail** [comment]
Looking into this, the auto-scrolling behavior during `Run All Cells` appears to be intentional and explicitly implemented.

In [NotebookActions.runAll](https://github.com/jupyterlab/jupyterlab/blob/main/packages/notebook/src/actions.tsx#L817), the notebook is forced to activate the last cell:

```
notebook.activeCellIndex = lastIndex;
```

Setting `activeCellIndex` implicitly scrolls the notebook to reveal that cell, which explains why the view jumps to the bottom when running all cells, regardless of the current cursor position or selection. 
Commenting out this line prevents the auto-scrolling behavior.

This behavior is also documented in the function comment:
“The last cell in the notebook will be activated and scrolled into view.”


A possible improvement would be to make this behavior configurable, for example by introducing a notebook setting such as:
“Auto-scroll to last cell when running all cells” (enabled by default for backward compatibility)

Thoughts on this ? @krassowski @jtpio

**Comment by krassowski** [comment]
I think a first step would be adding an argument allowing to invoke the commands which run all or run cells up to selected without scrolling. I am not 100% sure what should be the default and it should have a setting.

**Comment by Meriem-BenIsmail** [comment]
Ok that makes sense. I could open a draft PR and we could discuss it more. Thanks for the quick reply @krassowski !

**Comment by Yahiewi** [comment]
Hey @Meriem-BenIsmail sorry I didn't see your comment when I started working on this.
I opened a draft PR for this that basically works exactly like you discussed here.
